src/protocol.rs

```diff
@@ -14,6 +14,8 @@ use std::{
 use unsigned_varint::{encode, decode};
 use crate::onion_addr::Onion3Addr;
 
+// All these values are obtained by converting hexadecimal protocol codes to u32.
+// The protocol codes are present in multiformats/multicodec repository.
```

Comment:
:+1: for documenting this.

I am not exactly sure what the source of truth is, though I think we should link to multiformats/multiaddr instead.

```suggestion
// Protocols as well as their corresponding codes are defined in
// https://github.com/multiformats/multiaddr/blob/master/protocols.csv .
```